src/Symfony/Component/Validator/Constraints/ImageValidator.php

```diff
@@ -43,6 +44,22 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
+        if ($constraint->strict) {
+            if (!function_exists('imagecreatefromstring')) {
+                throw new RuntimeException('Strict image validation requires installed and enabled GD extension');
+            }
+
+            $resource = @imagecreatefromstring(file_get_contents($value));
```

Comment:
Our use case: some people has uploaded images that cann't be resized or displayed in frontend.